src/Identity/Extensions.Core/src/IdentityBuilder.cs

```diff
@@ -21,6 +22,11 @@ public class IdentityBuilder
     /// <param name="services">The <see cref="IServiceCollection"/> to attach to.</param>
     public IdentityBuilder(Type user, IServiceCollection services)
     {
+        if (user.IsValueType)
+        {
+            throw new ArgumentException("User type can't be a value type.", nameof(user));
```

Comment:
Should we add tests for passing a ValueType to these APIs? To ensure they break if someone removes the checks.